module/platform/Webots/src/Webots.cpp

```diff
@@ -331,20 +332,18 @@ namespace module::platform {
             // Receiving
             read_io = on<IO>(fd, IO::READ | IO::CLOSE | IO::ERROR).then("Read Stream", [this](const IO::Event& event) {
                 if ((event.events & IO::READ) != 0) {
-                    // Service the watchdog
-                    emit<Scope::WATCHDOG>(ServiceWatchdog<Webots>());
                     // If we have not seen the welcome message yet, look for it
                     if (!connection_active) {
-                        // Initaliase the string with ???????
-                        std::string initial_message = std::string(7, '?');
-                        const int n                 = ::read(fd, initial_message.data(), sizeof(initial_message));
+                        // Initaliase the string with 0s
+                        std::array<char, 8> initial_message{};
+                        const int n = ::read(fd, initial_message.data(), initial_message.size());
```

Comment:
You know... the more I think about this, the more terrified I am that you'll get a random crash here sometime because the welcome message ends up without a null terminator and the comparison later goes on forever and crashes... Adding an extra null will keep us safe

```suggestion
                        // Initaliase the string with 0s
                        // make sure we have an extra character just in case we read something that isn't a null terminator
                        std::array<char, 9> initial_message{};
                        const int n = ::read(fd, initial_message.data(), initial_message.size() - 1);
```

Reply:
Unfortunate typo: "Initaliase"